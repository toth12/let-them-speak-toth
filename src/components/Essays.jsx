import React from 'react';
import Hero from './Hero';

const Essays = props => (
  <div>
    <Hero text='Essays' />
    <div className='container'>
      <blockquote>
      "Lorem ipsum dolor sit amet, arcu vulputate egestas dolor wisi, fugiat vestibulum etiam mattis sit, curabitur elit nulla vel, mus quis elit porttitor nulla."
      </blockquote>
      <p className='center-text'>&mdash;Source Attribution</p>
      <p>Lorem ipsum dolor sit amet, arcu vulputate egestas dolor wisi, fugiat vestibulum etiam mattis sit, curabitur elit nulla vel, mus quis elit porttitor nulla. Amet elit in convallis, metus nam at sed, ipsum phasellus a dapibus, ornare in massa pharetra. Montes elementum pede integer ac, pulvinar aliquam non in augue, orci nulla eros nulla. Vulputate ut sed eget eu, dolor aliquam velit quisque a. Fermentum venenatis rhoncus vitae congue, suspendisse potenti pellentesque vestibulum.</p>
      <p>Amet a. Tempor eu. Dignissim nunc. Morbi faucibus, fermentum urna in, vitae nulla habitasse. A sit, vehicula eleifend, per massa vitae. Amet a. Tempor eu. Dignissim nunc. Morbi faucibus, fermentum urna in, vitae nulla habitasse.</p>
      <Titles />
      <p>Lorem ipsum dolor sit amet, arcu vulputate egestas dolor wisi, fugiat vestibulum etiam mattis sit, curabitur elit nulla vel, mus quis elit porttitor nulla. Amet elit in convallis, metus nam at sed, ipsum phasellus a dapibus, ornare in massa pharetra. Montes elementum pede integer ac, pulvinar aliquam non in augue, orci nulla eros nulla. Vulputate ut sed eget eu, dolor aliquam velit quisque a. Fermentum venenatis rhoncus vitae congue, suspendisse potenti pellentesque vestibulum.</p>
      <p>Amet a. Tempor eu. Dignissim nunc. Morbi faucibus, fermentum urna in, vitae nulla habitasse. A sit, vehicula eleifend, per massa vitae. Amet a. Tempor eu. Dignissim nunc. Morbi faucibus, fermentum urna in, vitae nulla habitasse.</p>
    </div>
  </div>
)

const essayList = [
  {
    title: 'Lorem ipsum dolor sit amet, arcu vulputate',
    file: '/assets/files/sample.pdf',
  },
  {
    title: 'Ipsum dolor sit amet, arcu vulputate',
    file: '/assets/files/sample.pdf',
  },
  {
    title: 'Dolor sit amet, arcu vulputate',
    file: '/assets/files/sample.pdf',
  },
  {
    title: 'Amet a. Tempor eu. Dignissim nunc. Morbi faucibus',
    file: '/assets/files/sample.pdf',
  },
  {
    title: 'Tempor eu. Dignissim nunc. Morbi faucibus',
    file: '/assets/files/sample.pdf',
  },
  {
    title: 'Lorem ipsum dolor sit amet, arcu vulputate',
    file: '/assets/files/sample.pdf',
  },
  {
    title: 'A. Tempor eu. Dignissim nunc. Morbi faucibus',
    file: '/assets/files/sample.pdf',
  },
  {
    title: 'Tempor eu. Dignissim nunc. Morbi faucibus',
    file: '/assets/files/sample.pdf',
  }
];

const Titles = (props) => (
  <div className='essay-titles'>
    <h2>Table of Contents</h2>
    {essayList.map((t, idx) => (
      <a href={t.file}
        className='essay-title'
        key={idx}
        download>{idx + 1}. {t.title}
      </a>
    ))}
  </div>
)

export default Essays;