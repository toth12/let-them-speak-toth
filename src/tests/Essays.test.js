import React from 'react';
import Essays from '../components/Essays';
import { createBrowserHistory } from 'history'
import { Router } from 'react-router';
import renderer from 'react-test-renderer';

it('renders', () => {
  const tree = renderer
    .create(
      <Router history={createBrowserHistory()}>
        <Essays/>
      </Router>
    ).toJSON();
  expect(tree).toMatchSnapshot();
});