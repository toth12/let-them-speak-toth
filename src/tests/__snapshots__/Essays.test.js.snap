// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div>
  <div
    className="hero background-image"
  >
    <h1>
      Essays
    </h1>
  </div>
  <div
    className="container"
  >
    <blockquote>
      "Lorem ipsum dolor sit amet, arcu vulputate egestas dolor wisi, fugiat vestibulum etiam mattis sit, curabitur elit nulla vel, mus quis elit porttitor nulla."
    </blockquote>
    <p
      className="center-text"
    >
      â€”Source Attribution
    </p>
    <p>
      Lorem ipsum dolor sit amet, arcu vulputate egestas dolor wisi, fugiat vestibulum etiam mattis sit, curabitur elit nulla vel, mus quis elit porttitor nulla. Amet elit in convallis, metus nam at sed, ipsum phasellus a dapibus, ornare in massa pharetra. Montes elementum pede integer ac, pulvinar aliquam non in augue, orci nulla eros nulla. Vulputate ut sed eget eu, dolor aliquam velit quisque a. Fermentum venenatis rhoncus vitae congue, suspendisse potenti pellentesque vestibulum.
    </p>
    <p>
      Amet a. Tempor eu. Dignissim nunc. Morbi faucibus, fermentum urna in, vitae nulla habitasse. A sit, vehicula eleifend, per massa vitae. Amet a. Tempor eu. Dignissim nunc. Morbi faucibus, fermentum urna in, vitae nulla habitasse.
    </p>
    <div
      className="essay-titles"
    >
      <h2>
        Table of Contents
      </h2>
      <a
        className="essay-title"
        download={true}
        href="/assets/files/sample.pdf"
      >
        1
        . 
        Lorem ipsum dolor sit amet, arcu vulputate
      </a>
      <a
        className="essay-title"
        download={true}
        href="/assets/files/sample.pdf"
      >
        2
        . 
        Ipsum dolor sit amet, arcu vulputate
      </a>
      <a
        className="essay-title"
        download={true}
        href="/assets/files/sample.pdf"
      >
        3
        . 
        Dolor sit amet, arcu vulputate
      </a>
      <a
        className="essay-title"
        download={true}
        href="/assets/files/sample.pdf"
      >
        4
        . 
        Amet a. Tempor eu. Dignissim nunc. Morbi faucibus
      </a>
      <a
        className="essay-title"
        download={true}
        href="/assets/files/sample.pdf"
      >
        5
        . 
        Tempor eu. Dignissim nunc. Morbi faucibus
      </a>
      <a
        className="essay-title"
        download={true}
        href="/assets/files/sample.pdf"
      >
        6
        . 
        Lorem ipsum dolor sit amet, arcu vulputate
      </a>
      <a
        className="essay-title"
        download={true}
        href="/assets/files/sample.pdf"
      >
        7
        . 
        A. Tempor eu. Dignissim nunc. Morbi faucibus
      </a>
      <a
        className="essay-title"
        download={true}
        href="/assets/files/sample.pdf"
      >
        8
        . 
        Tempor eu. Dignissim nunc. Morbi faucibus
      </a>
    </div>
    <p>
      Lorem ipsum dolor sit amet, arcu vulputate egestas dolor wisi, fugiat vestibulum etiam mattis sit, curabitur elit nulla vel, mus quis elit porttitor nulla. Amet elit in convallis, metus nam at sed, ipsum phasellus a dapibus, ornare in massa pharetra. Montes elementum pede integer ac, pulvinar aliquam non in augue, orci nulla eros nulla. Vulputate ut sed eget eu, dolor aliquam velit quisque a. Fermentum venenatis rhoncus vitae congue, suspendisse potenti pellentesque vestibulum.
    </p>
    <p>
      Amet a. Tempor eu. Dignissim nunc. Morbi faucibus, fermentum urna in, vitae nulla habitasse. A sit, vehicula eleifend, per massa vitae. Amet a. Tempor eu. Dignissim nunc. Morbi faucibus, fermentum urna in, vitae nulla habitasse.
    </p>
  </div>
</div>
`;
